util/TLSProxy/Message.pm

```diff
@@ -198,14 +198,14 @@ sub get_messages
         print "  [".$record->decrypt_data."]\n";
     } elsif ($record->content_type == TLSProxy::Record::RT_ALERT) {
         my ($alertlev, $alertdesc) = unpack('CC', $record->decrypt_data);
-        #All alerts end the test
-        $end = 1;
         #A CloseNotify from the client indicates we have finished successfully
         #(we assume)
-        if (!$server && $alertlev == AL_LEVEL_WARN
+        if (!$end && !$server && $alertlev == AL_LEVEL_WARN
             && $alertdesc == AL_DESC_CLOSE_NOTIFY) {
             $success = 1;
         }
+        #All alerts end the test
+        $end = 1;
```

Comment:
That this was needed is probably a bug? I didn't investigate very carefully. It looks vaguely like https://boringssl.googlesource.com/boringssl/+/fa214e4a18785a4281375b56c00732b470e9cd5b.

The TLSProxy tests probably also should look at the exit code, but I wasn't sure how to do that. If I wrote this test with the bad record going client -> server, the tests don't notice. The server fails with non-zero exit code, but this code sees a client close_notify and assume the test succeeded.